### PR TITLE
feat: support multiple file uploads

### PR DESCRIPTION
**Pull request by SebassNoob**
This PR implements support for multiple uploads of ``Note``.

WIP and early in development!
image

### CODE REVIEW

**Comment by SebassNoob** [comment]
Current UI:

image

**Comment by SebassNoob** [comment]
personal todo:
- [ ] read fastapi docs
- [ ] implement backend changes
- [ ] remove temp logic and rewrite api req
- [ ] impl ui changes from kinji
- [ ] final review

**Comment by SebassNoob** [comment]
Current (working) ui:
image

**Comment by SebassNoob** [comment]
stuff to fix before review:
- [ ] restyle to peoples liking
- [ ] write tests >:(
- [ ] ready for review

File: holy-grail-backend/backend/app/api/endpoints/library.py [hunk 1]
```diff
@@ -15,25 +15,39 @@
...
+        
+            note = await Library.create(
+                session,
+                uploaded_file=form[f'file {i}'],
+                uploaded_by=authenticated.user_id,
+                data=form_data,
+                s3_bucket=s3_bucket,
+            )
+            
+            new_notes.append(await Library.get(session, note.id))
```
Comment (vichannnnn) [inline holy-grail-backend/backend/app/api/endpoints/library.py:48]: From the endpoint `response_model`, you can just directly get the list of  `NoteSchema` you created from the logic above, why need to await `Library.get` again?

Also, database queries like this are very expensive, don't do that

File: holy-grail-backend/backend/app/api/endpoints/library.py [hunk 2]
```diff
@@ -15,25 +15,39 @@
 notes_router = APIRouter()
 
 
-@router.post("/", response_model=NoteSchema)
+@router.post("/", response_model=list[NoteSchema])
```
Comment (vichannnnn) [inline holy-grail-backend/backend/app/api/endpoints/library.py:18]: Also, don't put the business and service logic in the endpoint layer, make a custom `classmethod` in your `Library` model called `create_multiple_notes` or something then do this logic there. 

The code here in the endpoint should just call the method and pass the `request` and `session` parameters in and that's it.

Also you need to handle for `document_name` that already exists, if one of the docs has name that already exists, make sure all the files passes the condition of upload before doing a batch upload.

File: holy-grail-backend/backend/app/api/endpoints/library.py [hunk 3]
```diff
@@ -15,25 +15,39 @@
...
 @conditional_rate_limit("5/minute")
 async def create_note(
     request: Request,
-    data: NoteCreateSchema = Depends(),
-    file: UploadFile = File(None),
     authenticated: Account = Depends(Authenticator.get_verified_user),
     session: AsyncSession = Depends(get_session),
     s3_bucket: boto3.client = Depends(get_s3_client),
 ):
```
[the remark this answers is not in the document]
Comment (SebassNoob) [reply holy-grail-backend/backend/app/api/endpoints/library.py:25]: Isnt this defined on line 18 already? Must include here as well or not

[the remark this answers is not in the document]
Comment (vichannnnn) [reply holy-grail-backend/backend/app/api/endpoints/library.py:25]: yea I saw the changes, before no `response_model` thats why

File: holy-grail-backend/backend/app/api/endpoints/library.py [hunk 4]
```diff
@@ -15,25 +15,37 @@
...
+        note_datas = [
+            NoteCreateSchema(
+                category=form[f"category {i}"],
+                subject=form[f"subject {i}"],
+                type=form[f"type {i}"],
+                document_name=form[f"document_name {i}"],
+            )
+            for i in range(max_index + 1)
+        ]
+
```
Comment (vichannnnn) [inline holy-grail-backend/backend/app/api/endpoints/library.py:39]: logic move to model CRUD class method to handle

**Review by vichannnnn — APPROVED** [review]
lgtm

**Pull request merged**